tests/cpp/test_resize.cpp

```diff
@@ -4051,6 +4083,436 @@ TEST_F(ResizeTest, SliceSliceConcatConcat) {
+  std::vector<c10::IValue> inputs({t0});
+
+  KernelExecutor ke;
+  ke.compile(&fusion, inputs);
+  auto outputs = ke.run(inputs);
+  testValidate(&fusion, outputs, inputs, __LINE__, __FILE__);
+}
+
+// Propagating slice to inputs with rehsape after slice
+TEST_F(ResizeTest, PropagateSliceToInputsWithReshape2) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({16, 100});
+
+  EnableOptionsGuard enable_options_guard;
+  EnableOptionsGuard::getCurOptions().set(EnableOption::IdModel, {"all"});
+
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+
+  auto tv2 = slice(
+      tv1,
+      {{fusion.zeroVal(), tv1->getLogicalDomain().at(0)->extent()},
+       {IrBuilder::create<Val>(1L), IrBuilder::create<Val>(50)}});
+
+  auto tv3 = reshape(tv2, {shape[0], 49}, {shape[0] * 49});
+
+  auto tv4 = set(tv3);
+
+  fusion.addOutput(tv4);
+
+  scheduler_tools::propagateResizeToInputs(tv2->definition());
+
+  auto ref_tv = tv4;
+
+  // Schedule the reference
+  ref_tv->flatten();
+  // For TIDx
+  ref_tv->split(0, 128);
+  // For BIDx
+  ref_tv->split(0, 4);
+
+  scheduler_tools::scheduleLoopDomainsLike(
+      fusion.allTvs(), ref_tv->getLoopDomain());
+
+  // Fusion should have a uniform loop domain
+  checkLoopDomainEquivalence(ref_tv);
+
+  inlineMost();
+
+  // All tensors, except for fusion inputs, should be fully inlined
+  for (auto tv : fusion.allTvs()) {
+    if (tv->isFusionInput()) {
+      continue;
+    }
+    EXPECT_EQ(tv->getComputeAtPosition(), tv->nDims());
+  }
+
+  ref_tv->axis(-1)->parallelize(ParallelType::TIDx);
+  ref_tv->axis(-2)->parallelize(ParallelType::BIDx);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto t0 = at::randn(shape, options);
+  std::vector<c10::IValue> inputs({t0});
+
+  KernelExecutor ke;
+  ke.compile(&fusion, inputs);
+  auto outputs = ke.run(inputs);
+  testValidate(&fusion, outputs, inputs, __LINE__, __FILE__);
+}
+
+TEST_F(ResizeTest, PropagateMultipleSlicesToInputs) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({-1, 100});
+
+  EnableOptionsGuard enable_options_guard;
+  EnableOptionsGuard::getCurOptions().set(EnableOption::IdModel, {"all"});
+
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+
+  auto tv2 = slice(
+      tv1,
+      {{fusion.zeroVal(), tv1->getLogicalDomain().at(0)->extent()},
+       {IrBuilder::create<Val>(1L), tv1->getLogicalDomain().at(1)->extent()}});
+
+  auto tv3 = slice(
+      tv2,
+      {{fusion.zeroVal(), tv2->getLogicalDomain().at(0)->extent()},
+       {IrBuilder::create<Val>(1L), tv2->getLogicalDomain().at(1)->extent()}});
+
+  auto tv4 = set(tv3);
+
+  fusion.addOutput(tv4);
+
+  // Propagate the first slice to tv1
+  scheduler_tools::propagateResizeToInputs(tv2->definition());
+
+  // Propagate the second slice to tv1 and tv2
+  scheduler_tools::propagateResizeToInputs(tv3->definition());
```

Comment:
My understanding is that if you did these propagations in another order, or if you omitted the first propagation, then you might wind up with unmapped domains because the tv2 Resizes must be producers for the tv3 Resizes. Are there cases where a valid propagation order can't be found? I am thinking of something like this:
```
tv1 = slice(tv0)
tv2 = slice(tv1)
tv3 = slice(tv1)
```
Propagating from either `tv2` or `tv3` does not propagate to the other so their loop domains will not be mapped. Is my understanding wrong or does this mean we cannot inline fusions like this?

Reply:
You're completely right. The ordering is important, and also if a single tensor is resized multiple times, there may be a conflict. My approach at this point is to recompute such tensors so that each resize has a unique input tensor. This may not be optimal, but I found it good enough for RoPE-like fusions.

Alternatively, instead of propagating the slice from tv2 and tv3, we could leave tv1 as is, which would mean tv1 would have a different loop domain as tv2 and tv3, and there would be a synchronization. However, as long as tv1 is small enough to be stored in shared memory, the synchronization would be just syncthreads, so it might be just fine.

Reply:
That makes sense!